### ISSUE TITLE
Bump glfw from 3.2 to 3.3

### ISSUE THREAD

**Issue by coderedart** [emscripten-core/emscripten#18729]
Glfw support is missing content scale option.
eg: `glfwGetWindowContentScale` 

This causes `glfwGetWindowSize` and `glfwGetFramebufferSize` to return the same size even on platforms with scale != 1.0. The same problem doesn't exist with sdl2 as upstream sdl2 uses `emscripten_get_device_pixel_ratio` and deals with this itself.

This glfw problem makes the canvas really blurry on hidpi screens due to the mismatch of raw pixels and canvas pixels.

**Version of emscripten/emsdk:**
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.26-git (8eaf19f1c6a9a1b0cd0f9a91657366829e34ae5c)
clang version 16.0.0 (/srcdest/llvm-project f81f0cb75a2808a67d2662f044ad07628fc9d900)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: /opt/emscripten-llvm/bin

**Comment by coderedart** [comment]
Content Scale was added in glfw 3.3, so this requires upgrading the glfw header.

**Comment by sbc100** [comment]
Would you have time to send a PR for this maybe?  If not maybe someone else who uses GLFW could take a look.

**Comment by coderedart** [comment]
I can try, but I will probably need some help. Is there a guide for contributing? :sweat_smile: I can't figure out how to build/test my changes.

**Comment by kripken** [comment]
These links might be useful @coderedart :

* https://emscripten.org/docs/contributing/contributing.html
* https://emscripten.org/docs/contributing/developers_guide.html#developer-s-guide

But in general the simplest thing for testing a change is to open a PR with it. Then the automatic tests will do everything for you. When you need to debug stuff then a local setup is important, but for bumping the glfw version hopefully things will just work.

**Comment by coderedart** [comment]
can anyone with permission allow workflows to run in https://github.com/emscripten-core/emscripten/pull/18818

The js files are full or errors due to c macros, so its hard for me to know of any issues when i am editing it.

**Comment by coderedart** [comment]
created a new PR https://github.com/emscripten-core/emscripten/pull/18826 

I think this should work because i build everything from scratch including llvm, and made sure to run tests :)

**Comment by coderedart** [comment]
blocked on clarification from https://github.com/glfw/glfw/issues/2282

atm, we send canvas size as width/height, but when device pixel ratio is not 1, we need to set canvas.style.width/height according to that ratio and send canvas style width/height values as window size, canvas width/height as framebuffer size. atleast, that's how i think it should work, but I am trying to ask glfw devs to confirm that.

**Comment by coderedart** [comment]
https://github.com/emscripten-core/emscripten/pull/18826 is ready for review now. 

whatever the problem about pixel ratio, it can be dealt with in a different PR. For now, i updated the header and added some tests to check that those functions atleast work.

**Comment by Rucadi** [comment]
Shouldn't this be closed since https://github.com/emscripten-core/emscripten/pull/18826 was merged?
